# Compare Randomised Call Datasets to Observed
Here we look at the significance of the transition probabilities observed in our recorded calls, by comparing these to the transition probabilities obtained in the 10, 000 simulations of randomly generated calls (sensu Monte Carlo simulations). Significance is calculated as the relative ranking of each observed statistic (transition probability) among the sample values from the Monte Carlo randomization. 

The observed transition probability was found to be significantly different from random if it occurs within the 95th percentile for an upper or lower tail of the Monte Carlo randomizations.

For example, in our dataset of recorded calls the probability to transition from a DS segment to another DS segment is 0.07. This statistic is deemed as significant if it is either greater or lower than at least 95% of the values for the DS to DS transition probabilities in the Monte Carlo randomizations).

In [1]:
from tqdm.auto import tqdm

In [2]:
import avgn

In [3]:
import pandas as pd
import numpy as np
from avgn.utils.paths import DATA_DIR, ensure_dir, FIGURE_DIR

In [4]:
from scipy.stats import kruskal

In [5]:
DATASET_ID = "git_repos"

In [6]:
DT_ID = '2022-03-04_18-41-29'

In [8]:
randdfs = pd.read_pickle(DATA_DIR / DATASET_ID / 'Monte_Carlo' /  'random_simulations.pickle')
randdfs[:3]

DS_DS     DS_LH     DS_NL     DS_SH     LH_DS     LH_LH     LH_NL  \
0  0.352113  0.140845  0.102113  0.404930  0.342342  0.108108  0.117117   
0  0.368030  0.144981  0.144981  0.342007  0.318182  0.154545  0.181818   
0  0.366300  0.113553  0.139194  0.380952  0.275862  0.206897  0.120690   

      LH_SH     NL_DS     NL_LH     NL_NL     NL_SH     SH_DS     SH_LH  \
0  0.432432  0.333333  0.175926  0.138889  0.351852  0.345725  0.193309   
0  0.345455  0.384615  0.134615  0.105769  0.375000  0.352941  0.169550   
0  0.396552  0.373626  0.186813  0.109890  0.329670  0.335616  0.157534   

      SH_NL     SH_SH  rand_run  
0  0.122677  0.338290         0  
0  0.089965  0.387543         1  
0  0.150685  0.356164         2

In [9]:
len(randdfs)

10000

# Original Transition Probs

In [10]:
trans_origdf = pd.read_pickle(DATA_DIR / DATASET_ID / 'Monte_Carlo' /  'observed_transitions.pickle')
trans_origdf

DS_DS     DS_LH  DS_NL  DS_SH  LH_DS  LH_LH  LH_NL  LH_SH     NL_DS  \
0  0.071429  0.178571    0.0   0.75    1.0    0.0    0.0    0.0  0.795918   

   NL_LH  NL_NL     NL_SH     SH_DS     SH_LH     SH_NL     SH_SH  
0    0.0    0.0  0.204082  0.487654  0.234568  0.006173  0.271605

## DS_DS

In [14]:
trans_origdf["DS_DS"].values

array([0.07142857])

In [15]:
x = 0.07142857

In [16]:
rand_prob = np.array(randdfs["DS_DS"].values)

In [17]:
min(rand_prob)

0.26515151515151514

In [18]:
max(rand_prob)

0.4496124031007752

In [19]:
# define calculation for lower tail significance
def min_monte_carlo(x, rand_prob, tot_samples):
    r = rand_prob <= x
    return np.divide((np.sum(r)+1), (tot_samples + 1))

In [20]:
##DS_DS
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [21]:
np.sum(rand_prob <= x)

0

In [22]:
#define calculation for upper tail significance
def max_monte_carlo(x, rand_prob, tot_samples):
    r = rand_prob >= x
    return np.divide((np.sum(r)+1), (tot_samples + 1))

In [23]:
##DS_DS
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [24]:
np.sum(rand_prob >= x)

10000

In [25]:
### All cases are greater than observed DS_DS transition probability

In [26]:
sig = {"Transition":["DS_DS"], "Lower Tail": [results_min], "Upper Tail": [results_max]}
sigdf = pd.DataFrame(sig)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS      0.0001         1.0

## DS_LH

In [27]:
trans_origdf["DS_LH"]

0    0.178571
Name: DS_LH, dtype: float64

In [28]:
x = 0.178571
rand_prob = np.array(randdfs["DS_LH"].values)

In [29]:
min(rand_prob)

0.08695652173913043

In [30]:
max(rand_prob)

0.22868217054263565

In [31]:
##DS_LH lower tail significance
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

0.9023097690230977

In [32]:
np.sum(rand_prob <= x)

9023

In [33]:
##DS_LH upper tail significance
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

0.09779022097790221

In [34]:
np.sum(rand_prob >= x)

977

DS_LH observed transition oustide the 95th percentile. There are 977 randomly generated transition probs (over 5% of 10000) greater than observed transition probability. 

In [36]:
sig = pd.DataFrame({"Transition":["DS_LH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779

## DS_NL

In [37]:
trans_origdf["DS_NL"]

0    0.0
Name: DS_NL, dtype: float64

In [38]:
min(randdfs["DS_NL"])

0.055762081784386616

In [39]:
max(randdfs["DS_NL"])

0.20652173913043478

In [40]:
x = 0
rand_prob = np.array(randdfs["DS_NL"].values)

In [41]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [42]:
np.sum(rand_prob <= x)

0

No values under 0 <- pretty obvious!

In [43]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [44]:
np.sum(rand_prob >= x)

10000

All values over 0

In [45]:
sig = pd.DataFrame({"Transition":["DS_NL"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000

## DS_SH

In [46]:
trans_origdf["DS_SH"]

0    0.75
Name: DS_SH, dtype: float64

In [47]:
min(randdfs["DS_SH"])

0.27058823529411763

In [48]:
max(randdfs["DS_SH"])

0.4708029197080292

In [49]:
x = 0.75
rand_prob = np.array(randdfs["DS_SH"].values)

In [50]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

1.0

In [51]:
np.sum(rand_prob <= x)

10000

In [52]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

9.999000099990002e-05

In [53]:
np.sum(rand_prob >= x)

0

In [54]:
#all random trans probs for DS_SH are below 0.75

In [55]:
sig = pd.DataFrame({"Transition":["DS_SH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010

## LH_DS

In [56]:
trans_origdf["LH_DS"]

0    1.0
Name: LH_DS, dtype: float64

In [57]:
min(randdfs["LH_DS"])

0.18421052631578946

In [58]:
max(randdfs["LH_DS"])

0.5135135135135135

In [59]:
x = 1
rand_prob = np.array(randdfs["LH_DS"].values)

In [60]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

1.0

In [61]:
np.sum(rand_prob <= x)

10000

In [62]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

9.999000099990002e-05

In [63]:
np.sum(rand_prob >= x)

0

In [64]:
#All rand probabilities for LH to DS are below observed transition probability

In [65]:
sig = pd.DataFrame({"Transition":["LH_DS"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010

## LH_LH

In [66]:
trans_origdf["LH_LH"]

0    0.0
Name: LH_LH, dtype: float64

In [67]:
min(randdfs["LH_LH"])

0.03389830508474576

In [68]:
max(randdfs["LH_LH"])

0.2773109243697479

In [69]:
x = 0
rand_prob = np.array(randdfs["LH_LH"].values)

In [70]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [71]:
np.sum(rand_prob <= x)

0

In [72]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [73]:
np.sum(rand_prob >= x)

10000

In [74]:
# All rnadom probs greater than observed for LH-LH

In [75]:
sig = pd.DataFrame({"Transition":["LH_LH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000

## LH_NL

In [76]:
trans_origdf["LH_NL"]

0    0.0
Name: LH_NL, dtype: float64

In [77]:
min(randdfs["LH_NL"])

0.026785714285714284

In [78]:
max(randdfs["LH_NL"])

0.2543859649122807

In [79]:
x = 0
rand_prob = np.array(randdfs["LH_NL"].values)

In [80]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [81]:
np.sum(rand_prob <= x)

0

In [82]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [83]:
np.sum(rand_prob >= x)

10000

In [84]:
## All rnadom probs greater than observed for LH-NL

In [85]:
sig = pd.DataFrame({"Transition":["LH_NL"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000

## LH_SH

In [86]:
trans_origdf["LH_SH"]

0    0.0
Name: LH_SH, dtype: float64

In [87]:
min(randdfs["LH_SH"])

0.20175438596491227

In [88]:
max(randdfs["LH_SH"])

0.5217391304347826

In [89]:
x = 0
rand_prob = np.array(randdfs["LH_SH"].values)

In [90]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [91]:
np.sum(rand_prob <= x)

0

In [92]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [93]:
np.sum(rand_prob >= x)

10000

In [94]:
### All rnadom probs greater than observed for LH-SH

In [95]:
sig = pd.DataFrame({"Transition":["LH_SH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000

## NL_DS

In [96]:
trans_origdf["NL_DS"]

0    0.795918
Name: NL_DS, dtype: float64

In [97]:
min(randdfs["NL_DS"])

0.17708333333333334

In [98]:
max(randdfs["NL_DS"])

0.5533980582524272

In [99]:
x = 0.795918
rand_prob = np.array(randdfs["NL_DS"].values)

In [100]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

1.0

In [101]:
np.sum(rand_prob <= x)

10000

In [102]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

9.999000099990002e-05

In [103]:
np.sum(rand_prob >= x)

0

In [104]:
### All rnadom probs lower than observed for NL_DS

In [105]:
sig = pd.DataFrame({"Transition":["NL_DS"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010

## NL_LH

In [106]:
trans_origdf["NL_LH"]

0    0.0
Name: NL_LH, dtype: float64

In [107]:
min(randdfs["NL_LH"])

0.04

In [108]:
max(randdfs["NL_LH"])

0.28888888888888886

In [109]:
x = 0
rand_prob = np.array(randdfs["NL_LH"].values)

In [110]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [111]:
np.sum(rand_prob <= x)

0

In [112]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [113]:
np.sum(rand_prob >= x)

10000

In [114]:
### All rnadom probs greater than observed for NL_LH

In [115]:
sig = pd.DataFrame({"Transition":["NL_LH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000

## NL_NL

In [116]:
trans_origdf["NL_NL"]

0    0.0
Name: NL_NL, dtype: float64

In [117]:
min(randdfs["NL_NL"])

0.030303030303030304

In [118]:
max(randdfs["NL_NL"])

0.26804123711340205

In [119]:
x = 0
rand_prob = np.array(randdfs["NL_NL"].values)

In [120]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [121]:
np.sum(rand_prob <= x)

0

In [122]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [125]:
np.sum(rand_prob >= x)

10000

In [126]:
### All rnadom probs greater than observed for NL_LH

In [127]:
sig = pd.DataFrame({"Transition":["NL_NL"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000

## NL_SH

In [128]:
trans_origdf["NL_SH"]

0    0.204082
Name: NL_SH, dtype: float64

In [129]:
min(randdfs["NL_SH"])

0.20224719101123595

In [130]:
max(randdfs["NL_SH"])

0.5625

In [131]:
x = 0.204082
rand_prob = np.array(randdfs["NL_SH"].values)

In [132]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

0.00019998000199980003

In [133]:
np.sum(rand_prob <= x)

1

In [134]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

0.9999000099990001

In [135]:
np.sum(rand_prob >= x)

9999

In [136]:
### All bar one random probs greater than observed for NL_SH

In [137]:
sig = pd.DataFrame({"Transition":["NL_SH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000
0      NL_SH     0.00020     0.99990

## SH_DS

In [138]:
trans_origdf["SH_DS"]

0    0.487654
Name: SH_DS, dtype: float64

In [139]:
min(randdfs["SH_DS"])

0.22580645161290322

In [140]:
max(randdfs["SH_DS"])

0.4639175257731959

In [141]:
x = 0.487654
rand_prob = np.array(randdfs["SH_DS"].values)

In [142]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

1.0

In [143]:
np.sum(rand_prob <= x)

10000

In [144]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

9.999000099990002e-05

In [145]:
np.sum(rand_prob >= x)

0

In [146]:
### All rnadom probs lower than observed for SH_DS

In [147]:
sig = pd.DataFrame({"Transition":["SH_DS"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000
0      NL_SH     0.00020     0.99990
0      SH_DS     1.00000     0.00010

## SH_LH

In [148]:
trans_origdf["SH_LH"]

0    0.234568
Name: SH_LH, dtype: float64

In [149]:
min(randdfs["SH_LH"])

0.08108108108108109

In [150]:
max(randdfs["SH_LH"])

0.22181818181818183

In [151]:
x = 0.235
rand_prob = np.array(randdfs["SH_LH"].values)

In [152]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

1.0

In [153]:
np.sum(rand_prob <= x)

10000

In [154]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

9.999000099990002e-05

In [155]:
np.sum(rand_prob >= x)

0

In [156]:
## All rnadom probs lower than observed for SH_LH

In [157]:
sig = pd.DataFrame({"Transition":["SH_LH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000
0      NL_SH     0.00020     0.99990
0      SH_DS     1.00000     0.00010
0      SH_LH     1.00000     0.00010

## SH_NL

In [158]:
trans_origdf["SH_NL"]

0    0.006173
Name: SH_NL, dtype: float64

In [159]:
min(randdfs["SH_NL"])

0.06761565836298933

In [160]:
max(randdfs["SH_NL"])

0.1992619926199262

In [161]:
x = 0.006173
rand_prob = np.array(randdfs["SH_NL"].values)

In [162]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

9.999000099990002e-05

In [163]:
np.sum(rand_prob <= x)

0

In [164]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

1.0

In [165]:
np.sum(rand_prob >= x)

10000

In [166]:
## All random probs higher than observed for SH_NL

In [167]:
sig = pd.DataFrame({"Transition":["SH_NL"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000
0      NL_SH     0.00020     0.99990
0      SH_DS     1.00000     0.00010
0      SH_LH     1.00000     0.00010
0      SH_NL     0.00010     1.00000

## SH_SH

In [168]:
trans_origdf["SH_SH"]

0    0.271605
Name: SH_SH, dtype: float64

In [169]:
min(randdfs["SH_SH"])

0.26199261992619927

In [170]:
max(randdfs["SH_SH"])

0.5

In [171]:
x = 0.271605
rand_prob = np.array(randdfs["SH_SH"].values)

In [172]:
##Lower tail sig
results_min = min_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_min

0.00019998000199980003

In [173]:
np.sum(rand_prob <= x)

1

In [174]:
##Upper tail sig
results_max = max_monte_carlo(x=x, rand_prob=rand_prob, tot_samples = 10000)
results_max

0.9999000099990001

In [175]:
np.sum(rand_prob >= x)

9999

In [176]:
## All bar one random probs higher than observed for SH_SH

In [177]:
sig = pd.DataFrame({"Transition":["SH_SH"], "Lower Tail": [results_min], "Upper Tail": [results_max]})
sigdf = pd.concat([sigdf, sig], axis = 0)
sigdf

Transition  Lower Tail  Upper Tail
0      DS_DS     0.00010     1.00000
0      DS_LH     0.90231     0.09779
0      DS_LH     0.90231     0.09779
0      DS_NL     0.00010     1.00000
0      DS_SH     1.00000     0.00010
0      LH_DS     1.00000     0.00010
0      LH_LH     0.00010     1.00000
0      LH_NL     0.00010     1.00000
0      LH_SH     0.00010     1.00000
0      NL_DS     1.00000     0.00010
0      NL_LH     0.00010     1.00000
0      NL_NL     0.00010     1.00000
0      NL_SH     0.00020     0.99990
0      SH_DS     1.00000     0.00010
0      SH_LH     1.00000     0.00010
0      SH_NL     0.00010     1.00000
0      SH_SH     0.00020     0.99990